test/Microsoft.Azure.Devices.Edge.Test.Common/IotHub.cs

```diff
@@ -76,14 +76,26 @@ public IotHub(string iotHubConnectionString, string eventHubEndpoint, Option<Uri
         public Task<Device> GetDeviceIdentityAsync(string deviceId, CancellationToken token) =>
             this.RegistryManager.GetDeviceAsync(deviceId, token);
 
+        public async Task<Device> CreateDeviceIdentityAsync(Device device, CancellationToken token)
```

Comment:
Should we make signature of CreateDeviceIdentityAsync and CreateEdgeDeviceIdentityAsync consistently to use "string deviceId"? We can create a private method with deviceId, isIotEdge, cancellationtoken; then currently these 2 methods will call this private method.  What do you think?

Reply:
If the caller can't pass in a `Device` object, then they'd need to pass in a lot more than just `string deviceId, bool isIotEdge, CancellationToken token`.  They'd also need to be able to supply `AuthenticationType authType, Option<X509Thumbprint> thumbprint, Option<string> scope`. At that point, it's easier just to ask the caller to build up a `Device` object and pass it in.

The signatures are different because we need a lot more flexibility in creating leaf devices than we do edge devices.